controllers/cluster_controller.go

```diff
@@ -592,7 +592,23 @@ func (r *FoundationDBClusterReconciler) getStatusFromClusterOrDummyStatus(logger
 		return status, nil
 	}
 
+	// When we reached this part of the code the above GetStatus() called failed for some reason.
 	if cluster.Status.Configured {
+		// If the cluster is currently under a version incompatible upgrade, we try to assume the current version based
+		// on the coordinator reachability. If all (or the majority) of coordinators are reachable with a specific version
+		// of fdbcli we can assume that the cluster is running with that version and update the cluster.Status.RunningVersion.
+		// In theory we could use the go bindings if they would expose that information from the multi-version bindings.
```

Comment:
Go version of this one? https://github.com/apple/foundationdb/blob/main/bindings/c/fdb_c.cpp#L683

Reply:
I haven't looked into the semantics of that method, but that might work. I'm not sure if `// Returns the protocol version reported by the coordinator this client is connected to` means that the connected coordinator also represents the quorum of running coordinators. Another subtle difference is that this will return the protocol version not the FDB release version (which could be mapped).